[Functionbeat] Cannot deploy pubsub function on GCP

With functionbeat pubsub function deployed to GCP, the deployment fails due to "Build failed". The configuration include nothing special and credentials looks fine.

## Info
For confirmed bugs, please report:
- Version: 7.6.1
- Operating System: Mac

## Config
```
# Configure which region your project is located in.
functionbeat.provider.gcp.location_id: "asia-northeast1"
# Configure which Google Cloud project to deploy your functions.
functionbeat.provider.gcp.project_id: "XXXXX"
# Configure the Google Cloud Storage we should upload the function artifact.
functionbeat.provider.gcp.storage_name: "br-functionbeat-deploy"

functionbeat.provider.gcp.functions:
  # Define the list of function availables, each function required to have a unique name.
  # Create a function that accepts events coming from Google Pub/Sub.
  - name: pubsub
    enabled: true
    type: pubsub

    # Description of the method to help identify them when you run multiples functions.
    description: "Google Cloud Function for Pub/Sub"

    # The maximum memory allocated for this function.
    # Default is 256MB.
    #memory_size: 256MB

    # Execution timeout in seconds. If the function does not finish in time,
    # it is considered failed and terminated. Default is 60s.
    #timeout: 60s

    # Email of the service account of the function. Defaults to {projectid}@appspot.gserviceaccount.com
    service_account_email: XXXXX-serviceaccount@XXXXX.iam.gserviceaccount.com

    # Labels of the function.
    #labels:
    # mylabel: label

    # VPC Connector this function can connect to.
    # Format: projects/*/locations/*/connectors/* or fully-qualified URI
    #vpc_connector: ""

    # Number of maximum instances running at the same time. Default is unlimited.
    #maximum_instances: 0

    trigger:
      resource: "projects/XXXXX/topics/br_pub_sub"
      #service: "pubsub.googleapis.com"

    # Define custom processors for this function.
    #processors:
    #  - dissect:
    #      tokenizer: "%{key1} %{key2}"

```

## Debug Messages
```
$ ./functionbeat -v -e -d "*" deploy pubsub
2020-03-05T16:03:30.158+0900	INFO	instance/beat.go:622	Home path: [/Users/yukiya/functionbeat-7.6.1-darwin-x86_64] Config path: [/Users/yukiya/functionbeat-7.6.1-darwin-x86_64] Data path: [/tmp] Logs path: [/tmp/logs]
2020-03-05T16:03:30.158+0900	DEBUG	[beat]	instance/beat.go:674	Beat metadata path: /tmp/meta.json
2020-03-05T16:03:30.158+0900	INFO	instance/beat.go:630	Beat ID: d1fd2bb0-6dd5-4515-a92c-13e787fc3952
2020-03-05T16:03:30.159+0900	DEBUG	[filters]	add_cloud_metadata/providers.go:126	add_cloud_metadata: starting to fetch metadata, timeout=3s
2020-03-05T16:03:33.160+0900	DEBUG	[filters]	add_cloud_metadata/providers.go:169	add_cloud_metadata: timed-out waiting for all responses
2020-03-05T16:03:33.160+0900	DEBUG	[filters]	add_cloud_metadata/providers.go:129	add_cloud_metadata: fetchMetadata ran for 3.000644591s
2020-03-05T16:03:33.160+0900	INFO	add_cloud_metadata/add_cloud_metadata.go:89	add_cloud_metadata: hosting provider type not detected.
2020-03-05T16:03:33.160+0900	DEBUG	[processors]	processors/processor.go:101	Generated new processors: add_host_metadata=[netinfo.enabled=[false], cache.ttl=[5m0s]], add_cloud_metadata=null
2020-03-05T16:03:33.162+0900	DEBUG	[cli-handler]	cmd/cli_handler.go:52	Starting deploy for: pubsub
2020-03-05T16:03:33.162+0900	DEBUG	[gcp]	gcp/cli_manager.go:42	Deploying function: pubsub
2020-03-05T16:03:33.162+0900	DEBUG	[provider]	gcp/template_builder.go:60	Compressing all assets into an artifact
2020-03-05T16:03:33.216+0900	DEBUG	[keystore]	keystore/keystore.go:90	Loading file keystore from /tmp/functionbeat.keystore
2020-03-05T16:03:34.334+0900	DEBUG	[provider]	gcp/template_builder.go:73	Compression is successful (zip size: 10241366 bytes)
2020-03-05T16:03:34.334+0900	DEBUG	[gcp.executor]	executor/executor.go:53	The executor is executing '4' operations for converging state
2020-03-05T16:03:34.334+0900	DEBUG	[gcp]	gcp/op_ensure_bucket.go:28	Verifying presence of Cloud Storage bucket: 'br-functionbeat-deploy'
2020-03-05T16:03:35.949+0900	DEBUG	[gcp]	gcp/op_ensure_bucket.go:44	Cloud Storage bucket created with name 'br-functionbeat-deploy', attributes: <nil>
2020-03-05T16:03:35.949+0900	DEBUG	[gcp]	gcp/op_upload_to_bucket.go:36	Uploading file 'pubsub' to bucket 'br-functionbeat-deploy' with size 10241366 bytes
2020-03-05T16:03:38.720+0900	DEBUG	[gcp]	gcp/op_upload_to_bucket.go:55	Upload to bucket was successful
2020-03-05T16:03:38.720+0900	DEBUG	[gcp]	gcp/op_create_function.go:47	Creating function projects/XXXXX/locations/asia-northeast1/functions/pubsub at gs://br-functionbeat-deploy/pubsub
2020-03-05T16:03:39.064+0900	DEBUG	[gcp]	gcp/op_create_function.go:66	Operation 'operations/ZWxhc3RpYy1zYS9hc2lhLW5vcnRoZWFzdDEvcHVic3ViL2RIT2R4UVdRMUNn' is in progress to create function projects/XXXXX/locations/asia-northeast1/functions/pubsub
2020-03-05T16:03:39.195+0900	DEBUG	[gcp]	gcp/op_wait_for_function.go:61	Operation operations/ZWxhc3RpYy1zYS9hc2lhLW5vcnRoZWFzdDEvcHVic3ViL2RIT2R4UVdRMUNn has not finished yet. Retrying in 5 seconds.
.....
2020-03-05T16:06:53.750+0900	DEBUG	[gcp]	gcp/op_wait_for_function.go:61	Operation operations/ZWxhc3RpYy1zYS9hc2lhLW5vcnRoZWFzdDEvcHVic3ViL2RIT2R4UVdRMUNn has not finished yet. Retrying in 5 seconds.
2020-03-05T16:06:58.850+0900	DEBUG	[gcp.executor]	executor/executor.go:76	The executor is rolling back previous execution, '3' operations to rollback
2020-03-05T16:06:58.988+0900	DEBUG	[gcp]	gcp/op_delete_file_bucket.go:34	Removing file 'pubsub' from bucket 'br-functionbeat-deploy'
2020-03-05T16:06:59.580+0900	DEBUG	[gcp]	gcp/op_delete_file_bucket.go:47	Successfully removed function 'pubsub' from bucket 'br-functionbeat-deploy'
2020-03-05T16:07:00.043+0900	DEBUG	[gcp.executor]	executor/executor.go:89	The rollback is successful
2020-03-05T16:07:00.043+0900	DEBUG	[gcp]	gcp/cli_manager.go:47	Deploy finish for function 'pubsub'
Function: pubsub, could not deploy, error: error while creating function (code: 3):
Build failed: # serverlessapp/vendor/pubsub/vendor/github.com/elastic/beats/libbeat/common/file
/tmp/sgb/gopath/src/serverlessapp/vendor/pubsub/vendor/github.com/elastic/beats/libbeat/common/file/helper_other.go:28:6: SafeFileRotate redeclared in this block
	previous declaration at /tmp/sgb/gopath/src/serverlessapp/vendor/pubsub/vendor/github.com/elastic/beats/libbeat/common/file/helper_aix.go:26:44

2020-03-05T16:07:00.043+0900	DEBUG	[cli-handler]	cmd/cli_handler.go:62	Deploy execution ended
Fail to deploy 1 function(s)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -166,6 +166,11 @@ jobs:
       env: TARGETS="TEST_ENVIRONMENT=0 -C x-pack/functionbeat testsuite"
       go: $TRAVIS_GO_VERSION
       stage: test
+    - os: linux
+      before_install: .ci/scripts/travis_has_changes.sh x-pack/functionbeat libbeat || travis_terminate 0
+      env: TARGETS="-C x-pack/functionbeat test-gcp-functions"
+      go: 1.13.1
+      stage: test
 
     # Docker Log Driver
     - os: linux

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -432,6 +432,9 @@ pipeline {
             stage('Functionbeat x-pack'){
               steps {
                 mageTarget("Functionbeat x-pack Linux", "x-pack/functionbeat", "update build test")
+                withEnv(["GO_VERSION=1.13.1"]){
+                  makeTarget("Functionbeat x-pack Linux", "-C x-pack/functionbeat test-gcp-functions")
+                }
               }
             }
             stage('Functionbeat Mac OS X x-pack'){

--- a/x-pack/functionbeat/Makefile
+++ b/x-pack/functionbeat/Makefile
@@ -9,3 +9,6 @@ ES_BEATS?=../../
 #
 include $(ES_BEATS)/dev-tools/make/xpack.mk
 
+.PHONY:
+test-gcp-functions:
+	mage testGCPFunctions

--- a/x-pack/functionbeat/Makefile
+++ b/x-pack/functionbeat/Makefile
@@ -9,6 +9,6 @@ ES_BEATS?=../../
 #
 include $(ES_BEATS)/dev-tools/make/xpack.mk
 
-.PHONY:
-test-gcp-functions:
+.PHONY: test-gcp-functions
+test-gcp-functions: mage
 	mage testGCPFunctions

--- a/x-pack/functionbeat/magefile.go
+++ b/x-pack/functionbeat/magefile.go
@@ -185,7 +185,7 @@ func BuildPkgForFunctions() error {
 	return nil
 }
 
-// TestGCPFunctions are used by Travis to test if the GCP functions can be built with
+// TestGCPFunctions are used by the CI to test if the GCP functions can be built with
 // the selected Go version.
 // The version is 1.13.1 (Ref: https://cloud.google.com/functions/docs/concepts/go-runtime)
 func TestGCPFunctions() error {

--- a/x-pack/functionbeat/magefile.go
+++ b/x-pack/functionbeat/magefile.go
@@ -185,6 +185,29 @@ func BuildPkgForFunctions() error {
 	return nil
 }
 
+// TestGCPFunctions are used by Travis to test if the GCP functions can be built with
+// the selected Go version.
+// The version is 1.13.1 (Ref: https://cloud.google.com/functions/docs/concepts/go-runtime)
+func TestGCPFunctions() error {
+	for _, f := range []string{"pubsub", "storage"} {
+		params := devtools.DefaultBuildArgs()
+		inputFiles := filepath.Join("provider", "gcp", f, f+".go")
+		params.InputFiles = []string{inputFiles}
+		params.Name = f
+		params.CGO = false
+		params.Env = map[string]string{
+			"GOOS":   "linux",
+			"GOARCH": "amd64",
+		}
+
+		err := devtools.Build(params)
+		if err != nil {
+			return fmt.Errorf("error while building %s for GCP: %+v", f, err)
+		}
+	}
+	return nil
+}
+
 // BuildSystemTestBinary build a binary for testing that is instrumented for
 // testing and measuring code coverage. The binary is only instrumented for
 // coverage when TEST_COVERAGE=true (default is false).

--- a/x-pack/functionbeat/manager/gcp/template_builder.go
+++ b/x-pack/functionbeat/manager/gcp/template_builder.go
@@ -21,7 +21,7 @@ import (
 )
 
 const (
-	runtime          = "go111"                            // Golang 1.11
+	runtime          = "go113"                            // Golang 1.11
 	archiveURL       = "gs://%s/%s"                       // path to the function archive
 	locationTemplate = "projects/%s/locations/%s"         // full name of the location
 	functionName     = locationTemplate + "/functions/%s" // full name of the functions

--- a/x-pack/functionbeat/manager/gcp/template_builder.go
+++ b/x-pack/functionbeat/manager/gcp/template_builder.go
@@ -21,7 +21,7 @@ import (
 )
 
 const (
-	runtime          = "go113"                            // Golang 1.11
+	runtime          = "go113"                            // Golang 1.13
 	archiveURL       = "gs://%s/%s"                       // path to the function archive
 	locationTemplate = "projects/%s/locations/%s"         // full name of the location
 	functionName     = locationTemplate + "/functions/%s" // full name of the functions

